lib/services/role.go

```diff
@@ -1413,6 +1460,75 @@ func matchDenyImpersonateCondition(cond types.ImpersonateConditions, impersonate
 	return false, nil
 }
 
+// matchAllowRoleImpersonateCondition matches an allow impersonate condition
+// specifically for role-only impersonation, where only roles are matched.
+func matchAllowRoleImpersonateCondition(ctx *impersonateContext, whereParser predicate.Parser, cond types.ImpersonateConditions, impersonateRoles []types.Role) (bool, error) {
+	// an empty set matches nothing
+	if len(cond.Users) == 0 && len(cond.Roles) == 0 {
+		return false, nil
+	}
+
+	// Role impersonation can never apply to users.
+	if len(cond.Users) != 0 {
+		return false, nil
+	}
+
+	// By this point, at least 1 role is guaranteed.
+	anyRole, err := parse.NewAnyMatcher(cond.Roles)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	for _, impersonateRole := range impersonateRoles {
+		if !anyRole.Match(impersonateRole.GetName()) {
+			return false, nil
+		}
+		// TODO:
```

Comment:
Do you plan to address this here prior to merging?

Reply:
This is inherited from `matchAllowImpersonateCondition` (from which this TODO was copied and pasted) and I'm inclined to say refactoring `impersonateContext` would be out of scope here.

It's more than a little cursed though.